packages/nimble-components/src/table/specs/table-columns-hld.md

```diff
@@ -407,6 +407,55 @@ Clients should be allowed to use arbitrary content for the display part of a hea
 </nimble-table>
 ```
 
+### Validation
```

Comment:
From the discussion yesterday I think we want the public API for table column to be anlalogous to the Table validity api.

ie. exposes a validity property with boolean flags and a checkValidity method

And we also want to be able to have the table observe changes to column validity (table just needs a boolean for this column), I think the easiest way to do that is to align with our other apis and have an observable columnInternals.validConfiguration boolean or something. Table already has notifiers wired up for the columnInternals observable properties and can know if there are changes.

Reply:
Per recent conversations about PR size, adding this infrastructure in a separate PR from the mapping column implementations might be helpful. I would be ok with a PR that adds this only for existing columns which are always valid, as long as it has a test that exercises how the table behaves when a column reports itself as invalid.

Reply:
I'll make this a separate PR that is submitted before working on the mapping column PR.